src/data/minors/ai.ts

```diff
@@ -0,0 +1,119 @@
+import { CollegeOrMajorRequirement, Course } from '../../requirements/types';
+import {
+  includesWithSingleRequirement,
+  courseMatchesCodeOptions,
+  ifCodeMatch,
+  courseMatchesCode,
+} from '../../requirements/checkers';
+import { AdvisorGroup } from '../../tools/advisors/types';
+import { lastNameRange } from '../../tools/advisors/checkers';
+
+const aiMinorRequirements: readonly CollegeOrMajorRequirement[] = [
+  {
+    name: 'Machine Learning',
+    description: 'Add Core Course 1',
+    source:
+      'https://prod.cis.cornell.edu/undergraduate-opportunities/minors/artificial-intelligence/ai-minor-requirements',
+    checker: includesWithSingleRequirement(
+      'CS 3780',
+      'ECE 3200',
+      'ECE 4200',
+      'ORIE 3741',
+      'ORIE 4741',
+      'STSCI 3740'
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+
+  {
+    name: 'Reasoning',
+    description: 'Add Core Course 2',
+    source:
+      'https://prod.cis.cornell.edu/undergraduate-opportunities/minors/artificial-intelligence/ai-minor-requirements',
+    checker: includesWithSingleRequirement('CS 3700'),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+    slotNames: ['Course'],
+  },
+  {
+    name: 'Human-AI Interaction',
+    description: 'Add Core Course 3',
+    //TODO: add INFO 3450 exception for students grad in dec 2024 or may 2025
```

Comment:
Add a space between // and TODO.
We should to a `requirement migrations` to accommodate different major requirements based on entry year. You can check out this [documentation](https://www.notion.so/Requirement-Migrations-8f1684c570aa490bbd4dae9b4bd7bd7f) for it! This would be under a **modification**
Something along the lines of 
![image](https://github.com/user-attachments/assets/09e093c9-5a7f-40a1-a61b-b470e8cd28ed)

then follow the instructors for `index.ts` and running the commands in your terminal, then you should be all set!

Reply:
perhaps same idea for CS 4780/CS3780

Reply:
thanks for the requirement migration details, hannah! yes - we would want to add something like this. make sure that the file for the AI minor has the _current_ requirements and the migration is for previous requirements